File: SEP_T1/Domain/Models/enums/StatusEnum.cs
using System.ComponentModel;

namespace Domain.Models.enums;
public enum StatusEnum
{
    [Description("If the status is undefined")]
    Undefined = 0,
    
    [Description("If the post is active")]
    Active = 1,
    
    [Description("If the post is inactive")]
    Sold = 2,
    
    [Description("If the post is frozen for a potential seller")]
    Frozen = 3
}

File: SEP_T3/src/main/java/via/sdj3/sep_t3/model/enums/StatusEnum.java
package via.sdj3.sep_t3.model.enums;

import lombok.Getter;

/**
 * Status Enum for tracking the status of a post
 */
@Getter
public enum StatusEnum
{
    /**
     * Undefined status enum.
     */
    Undefined(0),
    /**
     * Active status, seller is looking for buyer
     */
    Active(1),
    /**
     * Sold status, seller sold the item
     */
    Sold(2),
    /**
     * Frozen status, seller found a buyer
     */
    Frozen(3);

    private final int number;
    StatusEnum(int number)
    {
        this.number=number;
    }
}
